### ISSUE TITLE
Fix legacy filter

### ISSUE THREAD

**Issue by knstvk** [Haulmont/jmix-cuba#31]
### QA
Check that legacy filter component works as in CUBA.
Test project - [legacy-filter.zip](https://github.com/Haulmont/jmix-cuba/files/6010685/legacy-filter.zip)

**Comment by knstvk** [comment]
Now it throws an exception on attempt to add a condition:
```
Caused by: java.lang.ClassCastException: java.util.UUID cannot be cast to io.jmix.core.Entity
	at com.haulmont.cuba.gui.components.data.tree.DatasourceTreeItems.getParent(DatasourceTreeItems.java:37) ~[jmix-cuba-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.component.tree.TreeDataProvider.getParent(TreeDataProvider.java:138) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.widget.JmixTree.getParentItem(JmixTree.java:120) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.widget.JmixTree.expandItemWithParents(JmixTree.java:133) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.component.impl.TreeImpl.expand(TreeImpl.java:611) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at com.haulmont.cuba.gui.components.filter.addcondition.AddConditionWindow.expandTreeRoots(AddConditionWindow.java:100) ~[jmix-cuba-0.2.0-SNAPSHOT.jar:na]
	at com.haulmont.cuba.gui.components.filter.addcondition.AddConditionWindow.init(AddConditionWindow.java:78) ~[jmix-cuba-0.2.0-SNAPSHOT.jar:na]
	at com.haulmont.cuba.gui.components.AbstractWindow.init(AbstractWindow.java:114) ~[jmix-cuba-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.core.common.event.EventHub.publish(EventHub.java:170) ~[jmix-core-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.screen.Screen.fireEvent(Screen.java:124) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.screen.UiControllerUtils.fireEvent(UiControllerUtils.java:57) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.sys.ScreensImpl.fireScreenInitEvent(ScreensImpl.java:1333) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.sys.ScreensImpl.createScreen(ScreensImpl.java:230) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at com.haulmont.cuba.web.sys.CubaScreens.openWindow(CubaScreens.java:100) ~[jmix-cuba-0.2.0-SNAPSHOT.jar:na]
	at com.haulmont.cuba.gui.components.filter.AddConditionHelper.addCondition(AddConditionHelper.java:95) ~[jmix-cuba-0.2.0-SNAPSHOT.jar:na]
	at com.haulmont.cuba.gui.components.filter.FilterDelegateImpl.lambda$createControlsLayoutForGeneric$2(FilterDelegateImpl.java:386) ~[jmix-cuba-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.core.common.event.EventHub.publish(EventHub.java:170) ~[jmix-core-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.component.impl.AbstractComponent.publish(AbstractComponent.java:84) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.component.impl.ButtonImpl.buttonClicked(ButtonImpl.java:78) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at io.jmix.ui.widget.JmixButton.fireClick(JmixButton.java:77) ~[jmix-ui-0.2.0-SNAPSHOT.jar:na]
	at com.vaadin.ui.Button$1.click(Button.java:57) ~[vaadin-server-8.9.2-5-jmix.jar:8.9.2-5-jmix]
```

**Comment by owlet1997** [comment]
1. Window's of editing filter property names should be related to 'filter', for example as in Cuba filter:
![image](https://user-images.githubusercontent.com/45730069/107849142-b5977200-6e12-11eb-8e7b-1164ad2053ff.png)
2. There is no parameters for customization 'width' and 'hidden' properties of filter groups as in Cuba filter:
![image](https://user-images.githubusercontent.com/45730069/107849209-39515e80-6e13-11eb-9d10-d7bd31cb1a4e.png)
3. Checkbox 'Default for me' in filter configuration and menu item 'make default'  in settings are not available
![image](https://user-images.githubusercontent.com/45730069/107849286-afee5c00-6e13-11eb-8b0a-564cceea3975.png)
![image](https://user-images.githubusercontent.com/45730069/107849291-c3012c00-6e13-11eb-8ce5-a6b3755de93a.png)
4. Name of this button should be 'Search'
![image](https://user-images.githubusercontent.com/45730069/107849426-8a158700-6e14-11eb-9ddc-56529245d524.png)
5. The properties are available in Component Inspector but are highlighted as a mistake on the XML-descriptor
- applyTo
- property
- custom
- manualApplyRequired
- editable
- folderActionsEnabled
- useMaxResults
![image](https://user-images.githubusercontent.com/45730069/107849521-2dff3280-6e15-11eb-88d2-07f341fb6b85.png)
![image](https://user-images.githubusercontent.com/45730069/107849668-428ffa80-6e16-11eb-8f90-4edc2feab2eb.png)
![image](https://user-images.githubusercontent.com/45730069/107849719-a5819180-6e16-11eb-8bf6-626220d4e1cb.png)
![image](https://user-images.githubusercontent.com/45730069/107849767-e9749680-6e16-11eb-9ee5-bd1fe22aaced.png)
![image](https://user-images.githubusercontent.com/45730069/107849829-2fc9f580-6e17-11eb-8dbf-449a69e5bb0a.png)
![image](https://user-images.githubusercontent.com/45730069/107849853-6738a200-6e17-11eb-8453-343ab10e354e.png)
![image](https://user-images.githubusercontent.com/45730069/107849900-aebf2e00-6e17-11eb-920c-c16fb6474de3.png)

**Comment by owlet1997** [comment]
Build failed, there is an exception after run application:
```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'cuba_ScreensLinkHandlerProcessor' defined in URL [jar:file:/C:/Users/shatalovanv/.gradle/caches/modules-2/files-2.1/io.jmix.cuba/jmix-cuba/0.3.0
-SNAPSHOT/88eea682be54ecd32a98f10756bd61c890b17dd9/jmix-cuba-0.3.0-SNAPSHOT.jar!/com/haulmont/cuba/web/sys/linkhandling/ScreensLinkHandlerProcessor.class]: Post-processing of merged bean definition failed; nested exception is java.l
ang.IllegalStateException: Failed to introspect Class [com.haulmont.cuba.web.sys.linkhandling.ScreensLinkHandlerProcessor] from ClassLoader [sun.misc.Launcher$AppClassLoader@73d16e93]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:586) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:531) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:335) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:333) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:208) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:944) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:923) ~[spring-context-5.3.3.jar:5.3.3]
        at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:588) ~[spring-context-5.3.3.jar:5.3.3]
        at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:144) ~[spring-boot-2.4.2.jar:2.4.2]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:767) [spring-boot-2.4.2.jar:2.4.2]
        at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759) [spring-boot-2.4.2.jar:2.4.2]
        at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:426) [spring-boot-2.4.2.jar:2.4.2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:326) [spring-boot-2.4.2.jar:2.4.2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1311) [spring-boot-2.4.2.jar:2.4.2]
        at org.springframework.boot.SpringApplication.run(SpringApplication.java:1300) [spring-boot-2.4.2.jar:2.4.2]
        at com.company.poj.PojApplication.main(PojApplication.java:16) [main/:na]
Caused by: java.lang.IllegalStateException: Failed to introspect Class [com.haulmont.cuba.web.sys.linkhandling.ScreensLinkHandlerProcessor] from ClassLoader [sun.misc.Launcher$AppClassLoader@73d16e93]
        at org.springframework.util.ReflectionUtils.getDeclaredFields(ReflectionUtils.java:737) ~[spring-core-5.3.3.jar:5.3.3]
        at org.springframework.util.ReflectionUtils.doWithLocalFields(ReflectionUtils.java:669) ~[spring-core-5.3.3.jar:5.3.3]
        at org.springframework.orm.jpa.support.PersistenceAnnotationBeanPostProcessor.buildPersistenceMetadata(PersistenceAnnotationBeanPostProcessor.java:407) ~[spring-orm-5.3.3.jar:5.3.3]
        at org.springframework.orm.jpa.support.PersistenceAnnotationBeanPostProcessor.findPersistenceMetadata(PersistenceAnnotationBeanPostProcessor.java:388) ~[spring-orm-5.3.3.jar:5.3.3]
        at org.springframework.orm.jpa.support.PersistenceAnnotationBeanPostProcessor.postProcessMergedBeanDefinition(PersistenceAnnotationBeanPostProcessor.java:335) ~[spring-orm-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyMergedBeanDefinitionPostProcessors(AbstractAutowireCapableBeanFactory.java:1100) ~[spring-beans-5.3.3.jar:5.3.3]
        at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:583) ~[spring-beans-5.3.3.jar:5.3.3]
        ... 16 common frames omitted
Caused by: java.lang.NoClassDefFoundError: Lio/jmix/ui/exception/EntityAccessExceptionHandler;
        at java.lang.Class.getDeclaredFields0(Native Method) ~[na:1.8.0_202]
        at java.lang.Class.privateGetDeclaredFields(Class.java:2583) ~[na:1.8.0_202]
        at java.lang.Class.getDeclaredFields(Class.java:1916) ~[na:1.8.0_202]
        at org.springframework.util.ReflectionUtils.getDeclaredFields(ReflectionUtils.java:732) ~[spring-core-5.3.3.jar:5.3.3]
        ... 22 common frames omitted
Caused by: java.lang.ClassNotFoundException: io.jmix.ui.exception.EntityAccessExceptionHandler
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382) ~[na:1.8.0_202]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_202]
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349) ~[na:1.8.0_202]
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_202]
        ... 26 common frames omitted
```

**Comment by glebfox** [comment]
After these changes https://github.com/Haulmont/jmix-ui/issues/237, the **jmix-cuba** artifacts have not yet been uploaded to the repository.

**Comment by owlet1997** [comment]
After adding work repository, refreshing dependencies and running application there is an exception:
```
Exception in handlers:
java.lang.ClassCastException: com.sun.proxy.$Proxy223 cannot be cast to com.haulmont.cuba.gui.WindowManager
	at com.haulmont.cuba.gui.exception.GenericExceptionHandler.handle(GenericExceptionHandler.java:42)
	at io.jmix.ui.exception.ExceptionHandlers.handle(ExceptionHandlers.java:114)
	at io.jmix.ui.AppUI.error(AppUI.java:436)
	at com.vaadin.server.LegacyCommunicationManager.handleConnectorRelatedException(LegacyCommunicationManager.java:426)
	at com.vaadin.server.communication.ServerRpcHandler.handleInvocation(ServerRpcHandler.java:434)
	at com.vaadin.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:396)
	at com.vaadin.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:260)
	at com.vaadin.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:82)
	at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1578)
	at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:425)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:327)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:115)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:81)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:119)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:113)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:126)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:81)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:105)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:101)
	at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:92)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:149)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:147)
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:125)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:103)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:89)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:110)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:80)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:55)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:211)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:183)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:888)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1597)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by owlet1997** [comment]
1. 'Make default' option
TC:
- run app
- open browser window
- create any filter condition and save filter
- click in 'Settings' 'Make default'
- reopen this window
AR:
filter isn't shown:
![image](https://user-images.githubusercontent.com/45730069/108958971-2cdbca00-768d-11eb-97ea-2d2f95900796.png)
but in settings it looks like filter is applied:
![image](https://user-images.githubusercontent.com/45730069/108959047-467d1180-768d-11eb-9eb4-84ce08fbce6d.png)

ER: as in Cuba application filter should be shown:
![image](https://user-images.githubusercontent.com/45730069/108958531-8d1e3c00-768c-11eb-805f-3b88eeb04a78.png)

**Comment by owlet1997** [comment]
2. 'Pin applied' option:
TC:
- run app
- open browser window
- create any filter condition
- click in 'Settings' 'Pin applied'

AR:
nothing is happened

ER:
as in Cuba app filter should be applied:
![image](https://user-images.githubusercontent.com/45730069/108960288-1cc4ea00-768f-11eb-8400-01fb54958527.png)

**Comment by owlet1997** [comment]
3. What about FullTextSearch?
property cuba.gui.genericFilterFtsDetailsActionEnabled  and jmix.gui.genericFilterFtsDetailsActionEnabled don't matter, there isn't any checkbox like this:
![image](https://user-images.githubusercontent.com/45730069/108963227-4253f280-7693-11eb-938d-55e0df9c447b.png)

**Comment by owlet1997** [comment]
4. Option 'Make default' in Settings
- run app
- open browser window
- click in 'Settings' 'Edit'
- add name to filter and create any filter conditions
- activate checkbox 'Default for me'
- save filter
- reopen window

AR:
filter is not applied

ER:
as in Cuba app filter must be applied and shown

**Comment by owlet1997** [comment]
5. Option 'Global default' in Settings

- run app
- open browser window
- click in 'Settings' 'Edit'
- add name to filter and create any filter conditions
- activate checkboxes 'Available for all users' and 'Global default'
- save filter
- reopen window

AR:
filter is not applied, threr is an error:
![image](https://user-images.githubusercontent.com/45730069/108983034-e051b800-76a7-11eb-9010-e5b22cfd2ed7.png)

ER:
as in Cuba app filter must be applied and shown

**Comment by owlet1997** [comment]
Property `property`:

- open XML-descriptor of browser screen
- add property 'property' into tag filter like this:

![image](https://user-images.githubusercontent.com/45730069/108986728-e21d7a80-76ab-11eb-8c81-355d939f09d5.png)
- run app
- open browser window
- click 'Add search condition' -> created property

AR: the EntityPicker is not clickable
![image](https://user-images.githubusercontent.com/45730069/108986684-d7fb7c00-76ab-11eb-9027-91a2a9083853.png)

ER:
as in Cuba app there must be a combobox with available instances from JPQL-query:
![image](https://user-images.githubusercontent.com/45730069/108987436-b6e75b00-76ac-11eb-9b7a-060a7fe4e83b.png)

**Comment by owlet1997** [comment]
7. Option 'TextMaxResults'

- open XML-descriptor of browser screen
- add option `'textMaxResults'=true`' into tag filter 
- run app
- open browser window
- look at filter

AR:
there is combobox 
![image](https://user-images.githubusercontent.com/45730069/108991642-cae18b80-76b1-11eb-90ef-2055ff984fb7.png)

ER: 
as in Cuba app there should be textfield like this:
![image](https://user-images.githubusercontent.com/45730069/108991758-e9478700-76b1-11eb-9a4f-d830cbbecbca.png)

**Comment by GlebDurygin** [comment]
Will be fixed in #43
